docs/connectors/sqlserver.asciidoc

```diff
@@ -96,7 +96,451 @@ We welcome any feedback from a user to try the plug-in with database in managed
+
+[NOTE]
+====
+When the columns for a row's primary/unique key are updated, the value of the row's key has changed so Debezium will output _three_ events: a `DELETE` event and a link:#tombstone-events[tombstone event] with the old key for the row, followed by an `INSERT` event with the new key for the row.
+====
+
+[[delete-events]]
+===== Delete events
+
+So far we've seen samples of _create_ and _update_ events.
+Now, let's look at the value of a _delete_ event for the same table. Once again, the `schema` portion of the value will be exactly the same as with the _create_ and _update_ events:
+
+[source,json,indent=0,subs="attributes"]
+----
+{
+  "schema": { ... },
+  },
+  "payload": {
+    "before": {
+      "id": 1005,
+      "first_name": "john",
+      "last_name": "doe",
+      "email": "noreply@example.org"
+    },
+    "after": null,
+    "source": {
+      "version": "0.10.0.Alpha1",
+      "connector": "sqlserver",
+      "name": "server1",
+      "ts_ms": 1559730445243,
+      "snapshot": false,
+      "db": "testDB",
+      "schema": "dbo",
+      "table": "customers",
+      "change_lsn": "00000027:00000db0:0005",
+      "commit_lsn": "00000027:00000db0:0007"
+    },
+    "op": "d",
+    "ts_ms": 1559730450205
+  }
+}
+----
+
+If we look at the `payload` portion, we see a number of differences compared with the _create_ or _update_ event payloads:
+
+* The `op` field value is now `d`, signifying that this row was deleted
+* The `before` field now has the state of the row that was deleted with the database commit.
+* The `after` field is null, signifying that the row no longer exists
+* The `source` field structure has many of the same values as before, except the `ts_ms`, `commit_lsn` and `change_lsn` fields have changed
+* The `ts_ms` shows the timestamp that Debezium processed this event.
+
+This event gives a consumer all kinds of information that it can use to process the removal of this row.
+
+The SQL Server connector's events are designed to work with https://cwiki.apache.org/confluence/display/KAFKA/Log+Compaction[Kafka log compaction],
+which allows for the removal of some older messages as long as at least the most recent message for every key is kept.
+This allows Kafka to reclaim storage space while ensuring the topic contains a complete dataset and can be used for reloading key-based state.
+
+[[tombstone-events]]
+When a row is deleted, the _delete_ event value listed above still works with log compaction, since Kafka can still remove all earlier messages with that same key.
+But only if the message value is `null` will Kafka know that it can remove _all messages_ with that same key.
+To make this possible, Debezium's SQL Server connector always follows the _delete_ event with a special _tombstone_ event that has the same key but `null` value.
+
+[NOTE]
```

Comment:
https://issues.apache.org/jira/browse/KAFKA-3832 is resolved in Kafka 2.0.2, so this note can be removed. I'll pushe a commit for this.